test/Microsoft.ApplicationInsights.AspNetCore.Tests/TelemetryInitializers/AzureWebAppRoleEnvironmentTelemetryInitializerTest.cs

```diff
@@ -0,0 +1,115 @@
+﻿namespace Microsoft.ApplicationInsights.AspNetCore.Tests.TelemetryInitializers
+{
+    using System;
+    using System.Globalization;
+    using Helpers;
+    using Microsoft.ApplicationInsights.AspNetCore.TelemetryInitializers;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.AspNetCore.Http;
+
+    using Xunit;    
+
+    public class AzureWebAppRoleEnvironmentTelemetryInitializerTest
+    {
+        [Fact]
+        public void AzureWebAppRoleEnvironmentTelemetryInitializerSetsRoleName()
+        {
+            var telemetryItem = new EventTelemetry();
+
+            Environment.SetEnvironmentVariable("WEBSITE_SITE_NAME", "TestRoleName");
+            Environment.SetEnvironmentVariable("WEBSITE_HOSTNAME", "TestRoleInstanceName");
+
+            var context = HttpContextAccessorHelper.CreateHttpContextAccessor(new RequestTelemetry(), null);
+            var initializer = new AzureWebAppRoleEnvironmentTelemetryInitializer(context);
+            initializer.Initialize(telemetryItem);
+
+            Assert.Equal("TestRoleName", telemetryItem.Context.Cloud.RoleName);
+            Assert.Equal("TestRoleInstanceName", telemetryItem.Context.Cloud.RoleInstance);
+            Assert.Equal("TestRoleInstanceName", telemetryItem.Context.GetInternalContext().NodeName);
+
+            Environment.SetEnvironmentVariable("WEBSITE_SITE_NAME", null);
+            Environment.SetEnvironmentVariable("WEBSITE_HOSTNAME", null);
```

Comment:
Since setting and clearing the environment variable happens for each test it would be good to have test setup and teardown methods that do those two respective lines before and after each test automatically.